int32 senderID
SolverVar[] vars